packaging/snapd.mk

```diff
@@ -0,0 +1,175 @@
+# This makefile is designed to be invoked from a build tree, not from the
+# source tree. This is particularly suited for building RPM packages. The goal
+# of this makefile is to *reduce* the duplication, *avoid* unintended
+# inconsistency between packaging of snapd across distributions.
+#
+# -----------------------------------------------------
+# Interface between snapd.mk and distribution packaging
+# -----------------------------------------------------
+#
+# Distribution packaging must generate snapd.defines.mk with a set of makefile
+# variable definitions that are discussed below. This allows the packaging
+# world to define directory layout and build configuration in one place and
+# this makefile to simply obey and implement that configuration.
+
+include snapd.defines.mk
+
+# There are two sets of definitions expected:
+# 1) variables defining various directory names
+vars += bindir sbindir libexecdir mandir datadir localstatedir unitdir
+# 2) variables defining build options:
+#   with_test_keys: set to 1 to build snapd with test key built in
+#   with_apparmor: set to 1 to build snapd with apparmor support
+#   with_core_bits: set to 1 to build snapd with things needed for the core/snapd snap
+#   with_alt_snap_mount_dir: set to 1 to build snapd with alternate snap mount directory
+vars += with_test_keys with_apparmor with_core_bits with_alt_snap_mount_dir
+# Verify that none of the variables are empty. This may happen if snapd.mk and
+# distribution packaging generating snapd.defines.mk get out of sync.
+
+$(foreach var,$(vars),$(if $(value $(var)),,$(error $(var) is empty or unset, check snapd.defines.mk)))
+
+# ------------------------------------------------
+# There are no more control knobs after this point
+# ------------------------------------------------
+
+# Import path of snapd.
+import_path=github.com/snapcore/snapd
```

Comment:
I'm actually in favor of `foo = bar` rather than `foo=bar`. Find the former easier to real.